docs/src/main/asciidoc/reactive-routes.adoc

```diff
@@ -163,6 +163,189 @@ public class SimpleRoutes {
 ----
 <1> The `path` value is used as a prefix for any route method declared on the class where `Route#path()` is used. The `produces` value is used for content-based routing for all routes where `Route#produces()` is empty.
 
+=== Returning Unis
+
+In a reactive route, you can return a `Uni` directly:
+
+[source,java]
+----
+@Route(path = "/hello")
+Uni<String> hello(RoutingContext context) {
+    return Uni.createFrom().item("Hello world!");
+}
+
+@Route(path = "/person")
+Uni<Person> getPerson(RoutingContext context) {
+    return Uni.createFrom().item(() -> new Person("neo", 12345));
+}
+----
+
+Returning `Unis` is convenient when using a reactive client:
+
+[source,java]
+----
+@Route(path = "/mail")
+Uni<Void> sendEmail(RoutingContext context) {
+    return mailer.send(...);
+}
+----
+
+The item produced by the returned `Uni` can be:
+
+* a string - written into the HTTP response directly
+* a buffer - written into the HTTP response directly
+* an object - written into the HTTP response after having been encoded into JSON.
+The `content-type` header is set to `application/json` if not already set.
+
+If the returned `Uni` produces a failure (or is `null`), an HTTP 500 response is written.
+
+Returning a `Uni<Void>` produces a 204 response (no content).
+
+=== Returning results
+
+You can also return a result directly:
+
+[source, java]
+----
+@Route(path = "/hello")
+String helloSync(RoutingContext context) {
+    return "Hello world";
+}
+----
+
+Be aware, the processing must be **non-blocking** as reactive routes are invoked on the IO Thread.
+Otherwise, use the `blocking` attribute of the `@Route` annotation.
+
+The method can return:
+
+* a string - written into the HTTP response directly
+* a buffer - written into the HTTP response directly
+* an object - written into the HTTP response after having been encoded into JSON.
+The `content-type` header is set to `application/json` if not already set.
+
+=== Returning Multis
+
+A reactive route can return a `Multi`.
+The items are written one by one, in the response.
+The response `Transfer-Encoding` header is set to `chunked`.
+
+[source, java]
+----
+@Route(path = "/hello")
+Multi<String> hellos(RoutingContext context) {
+    return Multi.createFrom().items("hello", "world", "!");  // <1>
+}
+----
+1. Produces `["hello", "world", "!"]`
+
+The method can return:
+
+* a `Multi<String>` - the items are written one by one in a JSON Array.
```

Comment:
Hm, does it make sense to always produce a JSON array? It's not very consistent with `Uni<String>` and `String` return types.

Reply:
It the behavior used in other frameworks. 
For raw output, use buffer.

Reply:
Well, I'm talking about use cases like `io.quarkus.qute.TemplateInstance.publisher()`. 

WRT consistency - we don't encode the `String` and `Uni<String>` return values to JSON either...

Reply:
> It the behavior used in other frameworks.

What frameworks BWT? ;-)

Reply:
Spring Routes do this, I also remember that Micronaut does something similar. 

The ideal would be to check the "produces" attribute from the Route annotation. However, at the moment, we generate one method per "method" which can have multiple Route annotations.

Reply:
> Spring Routes do this...

Yes, it seems that they do this based on the media type.

> The ideal would be to check the "produces" attribute from the Route annotation. However, at the moment, we generate one method per "method" which can have multiple Route annotations.

That would make perfect sense. But until we implement this I don't think it makes any sense to use some default (JSON) for `Multi<String>` but no default for `Uni<String>` and `String`.

Reply:
@mkouba and I discussed it and we are going to use the same approach we use for SSE. I'm going to update the PR soon.